app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt

```diff
@@ -192,6 +195,12 @@ internal class InitializationActivity : AppCompatActivity() {
     private fun runWithPostNotificationsPermission(action: () -> Unit) {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             // Android 13 and above
+            if (checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS)
+                != PackageManager.PERMISSION_GRANTED
+            ) {
+                splashScreenActive = false
+                setContentView(R.layout.splash)
+            }
```

Comment:
I think this will already be checked by quick permissions and it seems like a workaround because quick permissions doesn't expose the right info. Do you think this could be simplified by inlining parts of quick permissions, so that the hasPermissionCheck only has to be done once?

Reply:
I see what you mean. I could implement a new parameter to pass a function to the quickpermission call to execute code before starting to request permissions.  
That would prevent a duplicate permission check. What about this?

Reply:
Yeah, I'd prefer this.

Reply:
Implemented in 167eeaf1c25a4eaffad2ec32639ee8b09fe2e85b